### PR TITLE
feat(elbv2): Accept imported load balancer and target groups

### PR DESCRIPTION
**Pull request by edisongustavo**
After CDK 2.65.0, it is  now possible to obtain metrics through the interfaces. This opens up the possibility of creating dashboards for the imported versions of these resources.

PS: The tests have such large diffs because of the extra indentation.  The code is very similar, I just added the extra factory (`lbResourcesFactory`), but git is confused here.

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

### CODE REVIEW

File: .projenrc.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const { awscdk, javascript, DependencyType } = require("projen");
 const { workflows } = require("projen/lib/github");
 
-const CDK_VERSION = "2.60.0";
+const CDK_VERSION = "2.65.0";
```
Comment (echeung-amzn) [inline .projenrc.js:4]: Changing the peer dependencies is a breaking change.

You can refer to https://github.com/cdklabs/cdk-monitoring-constructs/pull/303 for an example of how to handle this. Note:

- The commit message format
- The bumping of `majorVersion` in this file
- Updating the README to reflect the updated peer deps

**Review by echeung-amzn — CHANGES_REQUESTED** [review]
Consider rewording the PR title/commit message to be more descriptive of the customer value rather than the implementation (e.g., "feat(elbv2): accept imported load balancer and target groups"

File: .projenrc.js (hunk 1 above)
Comment (edisongustavo) [reply .projenrc.js:4]: Thanks. I was indeed missing those steps. I've fixed it in the newest commit.

- [ ] The commit message format
- [ ] The bumping of majorVersion in this file
- [ ] Updating the README to reflect the updated peer deps

File: lib/monitoring/aws-loadbalancing/LoadBalancerMetricFactory.ts [hunk 2]
```diff
@@ -1,14 +1,42 @@
...
+  INetworkTargetGroup,
 } from "aws-cdk-lib/aws-elasticloadbalancingv2";
 
 import { ApplicationLoadBalancerMetricFactory } from "./ApplicationLoadBalancerMetricFactory";
 import { NetworkLoadBalancerMetricFactory } from "./NetworkLoadBalancerMetricFactory";
 import { MetricFactory, MetricWithAlarmSupport } from "../../common";
 
+// It's not possible to use instanceOf with typescript interfaces, so we have to use type guards to differentiate
+// between the interfaces. As another problem, the LoadBalancer/TargetGroup for both Network and Application types
+// don't have distinguished fields that could be used to differentiate between both types, so I'm resorting to using
```
Comment (echeung-amzn) [inline lib/monitoring/aws-loadbalancing/LoadBalancerMetricFactory.ts:14]:
```suggestion
// don't have distinguished fields that could be used to differentiate between both types, so we resort to using
```

File: lib/monitoring/aws-loadbalancing/LoadBalancerMetricFactory.ts [hunk 3]
```diff
@@ -1,14 +1,42 @@
...
+// the name of the class below.
+//
+// Ideally the 2 interfaces would provide a specific field to distinguish both types.
+function isApplicationLoadBalancer(
+  loadBalancer: INetworkLoadBalancer | IApplicationLoadBalancer
+): loadBalancer is IApplicationLoadBalancer {
+  return loadBalancer.constructor.name.indexOf("Application") != -1;
```
Comment (echeung-amzn) [inline lib/monitoring/aws-loadbalancing/LoadBalancerMetricFactory.ts:21]: Prefer strict equality (i.e. `!==`).

**Review by echeung-amzn — CHANGES_REQUESTED** [review]
LGTM, just some minor comments and will need a rebase.

**Pull request merged**